deploy.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+# Script adapted from https://gist.github.com/domenic/ec8b0fc8ab45f39403dd
+
+set -e # Exit with nonzero exit code if anything fails
+
+SOURCE_BRANCH="master"
+TARGET_BRANCH="gh-pages"
+
+function doCompile {
+  crystal doc
+}
+
+# Pull requests and commits to other branches shouldn't try to deploy, just build to verify
+if [ "$TRAVIS_PULL_REQUEST" != "false" -o "$TRAVIS_BRANCH" != "$SOURCE_BRANCH" ]; then
+    echo "Skipping deploy; just doing a build."
+    doCompile
+    exit 0
+fi
+
+# Save some useful information
+REPO=`git config remote.origin.url`
+SSH_REPO=${REPO/https:\/\/github.com\//git@github.com:}
+SHA=`git rev-parse --verify HEAD`
+
+# Clone the existing gh-pages for this repo into out/
+# Create a new empty branch if gh-pages doesn't exist yet (should only happen on first deply)
+git clone $REPO out
+cd out
+git checkout $TARGET_BRANCH || git checkout --orphan $TARGET_BRANCH
+cd ..
+
+mkdir -p out/$SOURCE_BRANCH
+
+# Clean out existing contents
+rm -rf out/$SOURCE_BRANCH/**/* || exit 0
+
+# Run our compile script
+doCompile
+
+# Move results
+mv doc/* out/$SOURCE_BRANCH/
```

Comment:
Don't we want to place it in `out/doc/master`.

Also we probably want to check if this is a tag build, and if so build the tag docs.